### PR TITLE
Add types to DALIDataType that were missing

### PR DESCRIPTION
**Pull request by klecki**
All common arithmetic types should be deemed worthy to use.

Signed-off-by: Krzysztof Lecki <klecki@nvidia.com>

#### Why we need this PR?
- It fixes a bug DALIDataType that missed some of the key arithmetic types for no reason.

#### What happened in this PR?
 - Explain solution of the problem, new feature added.
DALIDataType was extended with all typical arithmetic types
 - What was changed, added, removed?
DALIDataType, python enum wrapper, some tests. Use of standard type aliases was introduced.
 - What is most important part that reviewers should focus on?
Backward compatibility maybe?
 - Was this PR tested? How?
CI
 - Were docs and examples updated, if necessary?
no

**JIRA TASK**: [DALI-XXXX]

### CODE REVIEW

**Comment by klecki** [comment]
!build

File: dali/pipeline/data/types.h [hunk 1]
```diff
@@ -324,37 +328,41 @@ DLL_PUBLIC inline bool IsValidType(const TypeInfo &type) {
...
-DALI_REGISTER_TYPE(float16, DALI_FLOAT16);
-DALI_REGISTER_TYPE(float, DALI_FLOAT);
-DALI_REGISTER_TYPE(double, DALI_FLOAT64);
-DALI_REGISTER_TYPE(bool, DALI_BOOL);
-DALI_REGISTER_TYPE(string, DALI_STRING);
-DALI_REGISTER_TYPE(DALIImageType, DALI_IMAGE_TYPE);
-DALI_REGISTER_TYPE(DALIDataType, DALI_DATA_TYPE);
-DALI_REGISTER_TYPE(DALIInterpType, DALI_INTERP_TYPE);
-DALI_REGISTER_TYPE(DALITensorLayout, DALI_TENSOR_LAYOUT);
+DALI_REGISTER_TYPE(NoType,            NoType,           DALI_NO_TYPE);
```
Comment (mzient) [inline dali/pipeline/data/types.h]: Why? You just made the code more prone to future errors that may result from incomplete or misspelled copy-paste-replace.

File: dali/pipeline/data/types_test.cc [hunk 2]
```diff
@@ -33,20 +33,24 @@ namespace {
   static constexpr size_t DUMMY_ARRAY_SIZE = 42;
 }
 
-#define TYPENAME_FUNC(type)                     \
+#define TYPENAME_FUNC(type, typename)           \
```
Comment (mzient) [inline dali/pipeline/data/types_test.cc]: Don't keywords as macro arguments - even when they happen to _not_ collide.

Comment (klecki) [reply dali/pipeline/data/types_test.cc]: Done

**Pull request merged**